internal/services/gitserver/gitserver_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2019 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gitserver
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"agola.io/agola/internal/services/config"
+	"agola.io/agola/internal/util"
+)
+
+const (
+	branchName = "master"
+	tagName    = "v1.0"
+)
+
+func createTag(t *testing.T, ctx context.Context, git *util.Git, committerTime time.Time) {
+	if _, err := git.Output(ctx, nil, "branch", "test"); err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	if _, err := git.Output(ctx, nil, "checkout", "test"); err != nil {
+		t.Fatalf("unexpected err: %v", err)
+	}
+
+	git.Env = append(git.Env, "GIT_COMMITTER_DATE='"+committerTime.Format(time.RFC3339)+"'")
```

Comment:
Since this is a pure env (not a shell variable that requires quoting) I think you can avoid adding the single quotes.
Can you try if something cleaner like this works?
```
git.Env = append(git.Env, "GIT_COMMITTER_DATE="+committerTime.Format(time.RFC3339))
```
or as an alternative using fmt.Sprintf:
```
git.Env = append(git.Env, fmt.Sprintf("GIT_COMMITTER_DATE=%s', committerTime.Format(time.RFC3339)))
```